st2auth/st2auth/backends/__init__.py

```diff
@@ -60,10 +66,22 @@ def get_backend_instance(name):
         try:
             kwargs = json.loads(backend_kwargs)
         except ValueError as e:
-            raise ValueError('Failed to JSON parse backend settings: %s' % (str(e)))
+            raise ValueError('Failed to JSON parse backend settings for backend "%s": %s' %
+                             (name, str(e)))
     else:
         kwargs = {}
 
     cls = manager.driver
-    cls_instance = cls(**kwargs)
+
+    try:
+        cls_instance = cls(**kwargs)
+    except Exception as e:
+        tb_msg = traceback.format_exc()
+        class_name = cls.__name__
+        msg = ('Failed to instantiate auth backend "%s" (class %s) with backend settings '
+               '"%s": %s' % (name, class_name, str(kwargs), str(e)))
+        msg += '\n\n' + tb_msg
+        exc_cls = type(e)
+        raise exc_cls(msg)
+
     return cls_instance
```

Comment:
That's fairly common pattern  (it's used across the code base) and I personally prefer it over putting it inside `try` block at the end.

And in this case, as you, said there is also no need to do that (or use `else` block) since we always throw an exception inside the except block.

And even if we removed the except block, `cls_instance` would never be unset (this couldn't would never be reached) because it means it would throw before ever getting to this line :)